lang/syntax/src/trees/generic/def.rs

```diff
@@ -76,6 +83,20 @@ impl<P: Phase> Decl<P> {
             Decl::Let(_) => DeclKind::Let,
         }
     }
+
+    /// Returns whether the declaration is the "main" expression of the module.
+    pub fn get_main(&self) -> Option<Let<P>> {
+        match self {
+            Decl::Let(tl_let) => {
+                if tl_let.is_main() {
```

Comment:
The `if` could be replaced by `and_then` here: https://doc.rust-lang.org/std/option/enum.Option.html#method.and_then

Reply:
I don't see exactly how. The `is_main()` returns a boolean, not an option.

Reply:
Sorry, I meant `bool::then` or `bool::then_some`: https://doc.rust-lang.org/std/primitive.bool.html#method.then

Reply:
Thanks! I have applied this and amended the last commit.